demisto_sdk/commands/download/downloader.py

```diff
@@ -330,31 +318,62 @@ def replace_uuids(self, string_to_write: str, uuid_dict: dict) -> str:
+        #  if the content item is playbook and list-file flag is true, we should download the
+        #  file via direct REST API because there are props like scriptName, that playbook from custom
+        #  content bundle don't contain
+        return bool(not self.list_files and re.search(PLAYBOOK_REGEX, file_name))
 
-        if not self.list_files and re.search(
-            INCIDENT_FIELD_FILE_NAME_REGEX, member_name
-        ):
-            string_to_write = self.handle_incidentfield(
-                string_to_write, scripts_id_name
+    def write_custom_content(
+        self, content_items_file_names: List[Tuple[str, str]], scripts_id_to_name: dict
+    ):
+        for content_item_as_string, file_name in content_items_file_names:
+            if self.should_download_playbook(file_name):
+                content_item_as_string = self.download_playbook_yaml(
+                    content_item_as_string
+                )
+            content_item_as_string = self.replace_uuids(
+                content_item_as_string, scripts_id_to_name, file_name
             )
+            file_name = self.update_file_prefix(file_name.strip("/"))
+            path = Path(self.custom_content_temp_dir, file_name)
+            try:
+                path.write_text(content_item_as_string)
 
-        if not self.list_files and re.search(PLAYBOOK_REGEX, member_name):
-            #  if the content item is playbook and list-file flag is true, we should download the
-            #  file via direct REST API because there are props like scriptName, that playbook from custom
-            #  content bundle don't contain
-
-            string_to_write = self.download_playbook_yaml(string_to_write)
-
-        if not self.list_files and re.search(LAYOUT_FILE_NAME__REGEX, member_name):
-            string_to_write = self.handle_layout(string_to_write, scripts_id_name)
-
-        return string_to_write
+            except Exception:
+                logger.exception(
+                    "encountered exception, trying to write with encoding=utf8"
+                )
+                path.write_text(content_item_as_string, encoding="utf8")
+
+    def find_uuids_in_content_item(self, tar: tarfile.TarFile):
+        scripts_id_to_name: dict = {}
+        content_items_file_names_tuple: List[Tuple[str, str]] = []
+        for file in tar.getmembers():
+            file_name: str = self.update_file_prefix(file.name.strip("/"))
+            file_path: str = str(Path(self.custom_content_temp_dir, file_name))
+
+            if not (extracted_file := tar.extractfile(file)):
+                raise FileNotFoundError(
+                    f"Could not extract files from tar file: {file_path}"
+                )
+            string_to_write = extracted_file.read().decode("utf-8")
+            file_name = file.name.lower().lstrip("/")
+            if file_name.startswith(
+                (PLAYBOOK, AUTOMATION, INTEGRATION, LAYOUT, INCIDENT)
+            ):
```

Comment:
we can use `find_type` here:
- less code duplication
- uses smarter approaches than just the name

Reply:
Im not sure its possible.
find_type method relies on path in the content structure.
here the files are extracted from a downloaded tar which have a different file path structure than content Packs.